### PR TITLE
fix #80 - Convert to for processing setting 

### LINKED ISSUE #80: Convert to for processing setting
Using the Convert to for processing setting converts the currency amount on the final invoice incorrectly. 

500TIR is converted to 30USD on checkout. 
The invoice is marked as receiving 30TIR instead of 500TIR or 30USD

### PR DESCRIPTION
**Pull request by badr-r**
Fixing #80 
adding `$order_currency` (converted value) to blockonomics_orders table in the database
retrieving the `$order_currency` in the callback and using to retrieve exchange rate for that currency then it to calculate the correct invoice value

### CODE REVIEW

File: modules/gateways/Blockonomics/Blockonomics.php [hunk 1]
```diff
@@ -308,6 +308,7 @@ public function createOrderTableIfNotExist() {
 							$table->integer('bits_payed');
 							$table->string('blockonomics_currency');
 							$table->primary('addr');
+							$table->string('order_currency');
```
Comment (DarrenWestwood) [inline modules/gateways/Blockonomics/Blockonomics.php:311]: Does not work for existing users. createOrderTableIfNotExist() does not update blockonomics_orders table during upgrades. An example of performing table upgrades can be seen here:
https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/2373690f0b0fed963f13276ffb6fd405b8255d98/modules/gateways/Blockonomics/Blockonomics.php#L217-L221

File: modules/gateways/callback/blockonomics.php [hunk 2]
```diff
@@ -80,7 +81,10 @@
 } else {
 	$paymentAmount = $order['value'];
 }
-
+$exchangerate = Capsule::table('tblcurrencies')
+->where('code', $order['order_currency'])
+->get();
+$paymentAmount = $paymentAmount / $exchangerate[0]->rate;
```
Comment (DarrenWestwood) [inline modules/gateways/callback/blockonomics.php:87]: $paymentAmount should be rounded to 2 decimal places. Not doing so causes the correct amount to be received but the order status does not change to PAID.

This design requires re-evaluation. The exchange rate may be required to be saved during checkout to ensure this does not change between the time of making the payment and the payment being confirmed and added to the order.

**Review by DarrenWestwood — CHANGES_REQUESTED** [review]

File: modules/gateways/callback/blockonomics.php (hunk 2 above)
Comment (shivaenigma) [reply modules/gateways/callback/blockonomics.php:87]: >This design requires re-evaluation. The exchange rate may be required to be saved during checkout to ensure this does not change between the time of making the payment and the payment being confirmed and added to the order.
This should be fine. Exchange rate don't change so much. So need to do extra effort here

>$exchangerate[0]->rate
Is this correct code ? Will it work if client currency is EUR and converttoForProcessing in not selected. In that case won't it take USD/EUR exchange rate incorrectly ?

**Review by DarrenWestwood — CHANGES_REQUESTED** [review]

File: modules/gateways/Blockonomics/Blockonomics.php [hunk 3]
```diff
@@ -308,12 +308,22 @@ public function createOrderTableIfNotExist() {
...
 							$table->index('id_order');
 						}
 				);
 			} catch (Exception $e) {
 					exit("Unable to create blockonomics_orders: {$e->getMessage()}");
 			}
+		}else if(!Capsule::schema()->hasColumn('blockonomics_bitcoin_orders', 'order_currency')){
```
[the remark this answers is not in the document]
Comment (agarzon) [reply modules/gateways/Blockonomics/Blockonomics.php:318]: > Old database name has been used here. `blockonomics_bitcoin_orders` should be `blockonomics_orders`.

Actually WHMCS uses the prefix `mod_` in their thirdparty table names, so I suggest to be  `mod_blockonomics_orders`.

**Pull request closed without merging**